### DISCUSSION TITLE
Feat: custom completion support

### DISCUSSION THREAD

**Discussion by lukejans** [aws/codewhisperer-command-line-discussions#55] [Feature request]
### Checks

- [X] I could not find an existing setting that solved my issue
- [X] I searched [the discussions in this repo](https://github.com/aws/codewhisperer-command-line-discussions/discussions?q=) and there are no duplicates of my feature request

### Description:
> CodeWhisperer to accept custom cli completions.

### Realization
I recently started using ***[arduino-cli](https://arduino.github.io/arduino-cli/0.35/command-line-completion/)*** tools and have realized I'm constantly typing `arduino-cli help` because I can't remember all the commands. I realized this to be a trend with many command line tools downloaded with homebrew. homebrew has a configuration that makes tab completion work for downloaded command line tools but CodeWhisperer doesn't notice the commands.

### Feature
CodeWhisperer to recognize external commands bundled with homebrew completions. more on ***[homebrew completions](https://docs.brew.sh/Shell-Completion)***.

### Current functionality

commands can be listed via `tab` completion through homebrew configuration but CodeWhisperer does not notice the commands therefore it will not show a list of completions.

<img width="1180" alt="Screenshot 2024-01-10 at 3 50 50 PM" src="https://github.com/aws/codewhisperer-command-line-discussions/assets/91230392/6be28698-2bb1-4b18-b6a2-ed2a2373febc">

**Comment by grant0417** [comment]
CodeWhisperer uses the https://github.com/withfig/autocomplete specs to give auto-suggestions. These are community maintained, if you want to contribute `arduino-cli` we would merge the pr.

**Reply by lukejans** [reply]
arduino-cli completion spec [pull-request](https://github.com/withfig/autocomplete/pull/2231)
